src/Test/ItemsTestCase.php

```diff
@@ -0,0 +1,270 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Libero\ContentApiBundle\Test;
+
+use AppendIterator;
+use Libero\ContentApiBundle\Exception\ItemNotFound;
+use Libero\ContentApiBundle\Exception\UnexpectedVersionNumber;
+use Libero\ContentApiBundle\Exception\VersionNotFound;
+use Libero\ContentApiBundle\Model\ItemId;
+use Libero\ContentApiBundle\Model\ItemListPage;
+use Libero\ContentApiBundle\Model\Items;
+use Libero\ContentApiBundle\Model\ItemVersion;
+use Libero\ContentApiBundle\Model\ItemVersionNumber;
+use PHPUnit\Framework\TestCase;
+use function fopen;
+use function fwrite;
+use function iterator_to_array;
+use function rewind;
+
+abstract class ItemsTestCase extends TestCase
+{
+    /** @var Items */
+    protected $items;
+
+    /**
+     * @before
+     */
+    final protected function setUpItems() : void
+    {
+        $this->items = $this->createItems();
+    }
+
+    /**
+     * @test
+     */
+    final public function it_is_items() : void
+    {
+        $this->assertInstanceOf(Items::class, $this->items);
+    }
+
+    /**
+     * @test
+     */
+    final public function it_adds_and_removes_items() : void
+    {
+        $this->items->add($item1 = $this->generateItemVersion('foo', 1));
```

Comment:
so adding is part of the contract in the end; it is reasonable to expect the whole API to be implemented for dummy scenarios, despite not being production-ready in how it handles errors